docs/Advanced/SQL Searches/SQL Sorting.md

```diff
@@ -0,0 +1,31 @@
+---
+publish: true
+---
+
+# Sorting
+
+<span class="related-pages">#advanced/sql-search</span>
+
+There is no default sorting for the SQL based queries, if you want to have a order you need to specify it.
+
+## Using  `ORDER BY`
+
+Syntax:
```

Comment:
Feedback from pshev in [Discord](https://discord.com/channels/686053708261228577/965681451297304596/1097212035114221668)

- I did an ORDER BY description ASC and things looked off,.  From ACME  the order was ?, then Profit then Feed, which seems wrong.  But in the text of ACME, Profit is actually "==Profit==" and so sorts above Feeds.  Correct but NOT what a user would expect.  Didn't try with old tasks querie

From Clare: The existing tasks `sort by description` cleans out a bunch of formatting before sorting.

https://github.com/obsidian-tasks-group/obsidian-tasks/blob/309f73967df9a94d61521f40473f623268dc1740/src/Query/Filter/DescriptionField.ts#L57-L77

If cleaning out descriptions for sorting in SQL, please call the same function.

Reply:
not sure how to don that in a task-sql block as don't know obsidian api structure to access a plugin's functions.    ORDER BY cleanDescription(description) doesn't work.

Reply:
Oh sorry to have wasted your time @realpshev. I meant that for the author @sytone

Reply:
@claremacrae looking through the code it looks like the Task description is not what is being sorted here but a manipulated one with a set of data being replaced. It is the only field where the query is not actually filtering based on the contents of the backing field. 

In the scenario that @realpshev highlights here the description on the tasks side is being stripped of the Markdown metadata and only sorting based on what the user sees which is the rendered string. I spent some time updating the demo vault to try all these out and see where other issues may arise.

To replicate the current sorting I have added a function so users can choose to sort by the actual description or the rendered version by stripping the markdown from the string.

This will replicate the sorting and I have added a test to validate this scenario.

`ORDER BY removeMarkdown(description)`

Reply:
Found similar manipulation in the status field, added functions to replicate that so user can sort/filter on the string value or the internal allocation of type information or sorting. Added unit tests for these as well.

Reply:
These are all in the current PR now